### ISSUE TITLE
New Line Chart: WAU (Weekly Active Users) by Role

### ISSUE THREAD

**Issue by sarvaiyanidhi** [rubyforgood/casa#5589]
## What type(s) of user does this feature affect?
 - supervisors?
 - admins?

## Description
On the /health page, add a new line chart.
-WAU: count unique users by role (Volunteer, Supervisor, Admin) who logged in a given week on that month

Create a json endpoint for the chart data. Make sure the json contains the minimum amount of data for the chart. Make sure the endpoint is accessible to users who are logged out.

If chart creation fails, display an error to the user explaining why. Print a detailed version of the error in the console.

No testing required but prove your solution works.

See:

https://github.com/rubyforgood/casa/blob/main/app/controllers/health_controller.rb
https://github.com/rubyforgood/casa/blob/main/app/javascript/src/display_app_metric.js

We use [chart.js](https://www.chartjs.org/docs/latest/samples/line/line.html) for graphs

See: https://github.com/rubyforgood/casa/wiki/Working-with-Frontend-Assets

More details in issue #5530 

## How to access the QA site
_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in [slack](https://join.slack.com/t/rubyforgood/shared_invite/zt-21pyz2ab8-H6JgQfGGI0Ab6MfNOZRIQA) #casa channel to ask questions quickly. And [discord](https://discord.gg/qJcw2RZH8Q) for office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

**Comment by sarvaiyanidhi** [comment]
WIP

**Comment by sarvaiyanidhi** [comment]
WIP

**Comment by sarvaiyanidhi** [comment]
Raised PR for related issue #5652

**Comment by compwron** [comment]
Cool :)

**Comment by sarvaiyanidhi** [comment]
Hey @bcastillo32 ,

Just a quick update that I have raised separate PR which implements functionality to track each login of user. Once that PR is merged in main code I will continue further to implement MAU and WAU functionality asap.

**Comment by sarvaiyanidhi** [comment]
WIP.. Fixed MAU chart by fetching details now from login_activity table and merged with separate PR. Working on WAU chart.

**Comment by bcastillo32** [comment]
> WIP.. Fixed MAU chart by fetching details now from login_activity table and merged with separate PR. Working on WAU chart.

Thank you @sarvaiyanidhi  very much so looking forward to this one :)

**Comment by sarvaiyanidhi** [comment]
Sure @bcastillo32 Will try to close this asap. Thanks

**Comment by sarvaiyanidhi** [comment]
Will be working on this issue in this week and try to close it.

**Comment by sarvaiyanidhi** [comment]
Work in progress

**Comment by sarvaiyanidhi** [comment]
Was busy but will try to close and push PR soon.

**Comment by sarvaiyanidhi** [comment]
Long time pending.. Will close in this week.

**Comment by sarvaiyanidhi** [comment]
Restarted working on this issue..

**Comment by sarvaiyanidhi** [comment]
Planning to start on this next week.

**Comment by sarvaiyanidhi** [comment]
Work is in porgress

**Comment by sarvaiyanidhi** [comment]
Could not finish it due to some issues, will try to resume mid of next week and close it.

**Comment by bcastillo32** [comment]
> Could not finish it due to some issues, will try to resume mid of next week and close it.

Hi @sarvaiyanidhi I hope you are well. Will you be able to complete this? :)

**Comment by sarvaiyanidhi** [comment]
Hi @bcastillo32 I am good.. Hope all is well with you as well.

Sorry, this issue is pending from very long time and I would definitely like to take ownership and close it.

I need to again go through the details and work on it. Also, I remember that last time when I started working on this I was also looking to clean code as the file displaying chart has become too long.

Can you please post once again requirement details just for clarification here or in new issue so I can again resume it. Probably, I will try to work it over the weekend or next week.

Thanks
